### DISCUSSION TITLE
如何给Form.Array的其中一项赋值

### DISCUSSION THREAD

**Discussion by liuyongming1** [ant-design/ant-design-mobile#5438] [Q&A]
![微信图片_20220705105829](https://user-images.githubusercontent.com/98137206/177241401-d1c3e848-17c0-4b38-9d7a-3ca0790c5a30.png)
当点击修改的时候可以修改Form表单中的姓名，但是这个是通过Form.Array写的，如何给他进行赋值

**Comment by yaogengzhu** [comment]
这个文档看下： https://mobile.ant.design/zh/components/form#forminstance  看下是否对你有帮助

**Reply by liuyongming1** [reply]
我用setFieldsValue确实可以赋值，但是Form.Array是数组，好像没法用这种方式进行赋值，可能是我方式不对，我试了很多都不行

**Reply by qing-rain** [reply]
> 我用setFieldsValue确实可以赋值，但是Form.Array是数组，好像没法用这种方式进行赋值，可能是我方式不对，我试了很多都不行

数组我是用form.setFields用这个赋值的

**Comment by Lee-zee** [comment]
同样的场景，点击删除删除指定Form.Array下的表单项，setFieldValue键名传入[index,'name']无效

**Reply by qing-rain** [reply]
setFieldValue要完整的字段路径，要把Form.Array 的name一起传吧。字段路径是动态的话，用setFields比较好操作。
